### DISCUSSION TITLE
Question about python

### DISCUSSION THREAD

**Discussion by martinlombana** [c-smile/sciter-js-sdk#56] [General]
Hi!

If we want to run sciter-js inside python, like with https://github.com/sciter-sdk/pysciter, we should just exchange the dll and use the libsciter-gtk.so (in linux) or the dll for windows? Would that work out of the box?

Thanks!

**Comment by c-smile** [comment]
It should work OK as Sciter and Sciter.JS are ABI compatible. 
At least that's the intention so let me know if something will not work.

**Reply by AlecHaring** [reply]
I've found that Sciter.JS works well with PySciter for the most part aside from a few small issues. The largest compatibility issue I've stumbled upon is PySciter's ability to display errors that were thrown by Sciter.JS. For example, if you call `self.eval_script(...)` from Python and an error occurs, sciter.error.ScriptException is thrown but the sciter.Value object's value is "[object]", leaving the Python side with no information as to what happened.

![Screen Shot 2021-04-17 at 6 36 36 PM](https://user-images.githubusercontent.com/19616756/115128450-d9f70280-9fab-11eb-990a-cf7619253b62.png)

**Reply by c-smile** [reply]
It returns JavaScript Error object so you should be able to read its `message` property.

**Reply by pravic** [reply]
So, it's not `UT_STRING_ERROR` anymore?

**Reply by c-smile** [reply]
It should be, fixing

**Comment by martinlombana** [comment]
OK!. And, as for the raspberry pi, it would be the same? Or is there any thing else involved? I am receiving next week a rapberry pi 4, and I wanted to create a test with pysciter, sciter-js and pyinstaller (for packaging, portability).

**Comment by c-smile** [comment]
ARM32 version uses the same API as others.

**Comment by rjcreluda** [comment]
> 
> 
> Hi!
> 
> If we want to run sciter-js inside python, like with https://github.com/sciter-sdk/pysciter, we should just exchange the dll and use the libsciter-gtk.so (in linux) or the dll for windows? Would that work out of the box?
> 
> Thanks!

I tried that and it didn't work.
OSError: exception: access violation writing 0x0000000000000000
The following line caused the error:
frame = sciter.Window(ismain=True, uni_theme=True)
![Capture](https://user-images.githubusercontent.com/70935355/116000667-de4d9c00-a5f9-11eb-9822-d6bc32c7ee9a.PNG)

**Reply by pravic** [reply]
#85

**Reply by rjcreluda** [reply]
Thank you, it works well.
